### ISSUE TITLE
Integration with Orthanc DicomWeb with HTTP Authorization 

### ISSUE THREAD

**Issue by FossilBlade** [OHIF/Viewers#1597]
**Description**

**Is there any documentation which describes the steps to use OHIF viewer with Orthanc DicomWeb which has HTTP authentication ?**

I have given it couple of tries but it work only for the first view where the patient list is shown and fails when clicking on a patient to view the images.  It asks for the orthanc password. **My guess is OHIF Viewer is not sending the Authorization Header in this ajax request. (Snapshot at the bottom)**

I am using docker-compose for latest orhtanc-plugins and ohif viewer 

My use case is to use the viewer to show the user dicom images, who doesnt any idea about the orthanc as the backend. Also i can't disable orhtanc's authentication as that would be a security issue.

Any help would be appreciated. 

**Also is there a way so that the dicomweb user and password can be hidden as it is clearly visible in the app-config.js file which gets down loaded. Basically a better way to protect the dicomweb credentials.** 

Here is my docker-compose
```
version: '3.6'
services:

  viewer:
    image: ohif/viewer:v3.8.16.10895
    container_name: viewer
    depends_on:
      - orthanc
    restart: always
    ports:
      - '80:80'
    networks:
      - frontend    
    volumes:
      # Nginx as a reversed proxy for orthanc and the viewer
      - ./config/nginx.conf:/etc/nginx/conf.d/default.conf:ro
      # OHIF Viewer configuration to reach DICOMWeb endpoints
      - ./config/viewer.js:/usr/share/nginx/html/app-config.js:ro

  orthanc:
    image: jodogne/orthanc-plugins:1.6.0
    container_name: orthanc    
    restart: always
    networks:
      - frontend     
    volumes:      
      - ./config/orthanc.json:/etc/orthanc/orthanc.json:ro      
    command: /etc/orthanc/orthanc.json

networks:
  frontend:
```

Here is the relevant orthanc config:
```
"AuthenticationEnabled" : true,
  "RegisteredUsers" : {
    "final1":"final1"
  },

  "DicomWeb" : {
    "Enable" : true,            // Whether DICOMweb support is enabled
    "Root" : "/dicom-web/",     // Root URI of the DICOMweb API (for QIDO-RS, STOW-RS and WADO-RS)
    "EnableWado" : true,        // Whether WADO-URI (previously known as WADO) support is enabled
    "WadoRoot" : "/wado",       // Root URI of the WADO-URI (aka. WADO) API
    "Ssl" : false,              // Whether HTTPS should be used for subsequent WADO-RS requests
    "QidoCaseSensitive" : true, // For QIDO-RS server, whether search is case sensitive (since release 0.5)
    "StudiesMetadata" : "Full", // How study-level metadata is retrieved (since release 1.1, cf. section below)
    "SeriesMetadata" : "Full"   // How series-level metadata is retrieved (since release 1.1, cf. section below)
  }
```

Here is the viewer.js dicom part

```
dicomWeb: [
        {
          name: 'ORTHANC',
          wadoUriRoot: 'http://localhost:80/pacs/wado',
          qidoRoot: 'http://localhost:80/pacs/dicom-web',
          wadoRoot: 'http://localhost:80/pacs/dicom-web',
          qidoSupportsIncludeField: true,
          imageRendering: 'wadors',
          thumbnailRendering: 'wadors',
          enableStudyLazyLoad: true,
	  	  requestOptions: {
		          auth: "demo:demo",
		          logRequests: true,
		          logResponses: false,
		          logTiming: true,
		          requestFromBrowser: true,
		        }
        },
      ],
```

Screenshot of send page:

![image](https://user-images.githubusercontent.com/32032862/78329439-34f6cb00-75de-11ea-8625-d736c032c12d.png)

**Comment by swederik** [comment]
You should not use HTTP Basic Auth unless you store your credentials on a server side somewhere.

Did you look at https://docs.ohif.org/deployment/recipes/user-account-control.html ?

**Comment by FossilBlade** [comment]
Thanks for the reply @swederik 

I was looking at it in the documentation however the documentation didn't follow the project/folder structure hence i dint pursue it further.  The documentation requires few update to run these examples as per the current project/folder structure. But i ll try to run it today by taking hints from the docs.

However the bigger problem still exists, of the ajax call request, to get the the dicom frame, doesn't have the authorization header as shown in the the screen shot at the bottom of my question which is causing a 401 Unauthorized error. The same is being faced by other user for the latest version as described in #1604

**Comment by dannyrb** [comment]
> I was looking at it in the documentation however the documentation didn't follow the project/folder structure hence i dint pursue it further. The documentation requires few update to run these examples as per the current project/folder structure. But i ll try to run it today by taking hints from the docs.

Always happy to review and merge PRs for inconsistencies in docs ^_^

> The same is being faced by other user for the latest version as described in #1604

Have you confirmed this is a regression? Does your config work with older Docker images?

**Comment by FossilBlade** [comment]
Hey @dannyrb 

So I managed to step up Keycloak, nginx, orthanc, and OHIF viewer. However, there were issues with the OHIF Viewer. Viewer is able to redirect to the keycloak login page, however when login is successful viewer is not adding the Authorization header (from the keycloak access_token) to the AJAX request it is making to orthanc DicomWeb urls. I have attached the screen shot below. 

I can successfully login into orthanc explorer using the keycloak authentication but not OHIF.

Once my setup works successfully, I will create a PR for the keycloak recipe and the related documentation :)
 
**My observations can be reproduced using the repo i have created for running the setup locally:** https://github.com/FossilBlade/keycloak-nginx-viewer

**OHIF Viewer Redirection to keycloak login Page:**

![image](https://user-images.githubusercontent.com/32032862/78432513-f04c5c00-76d1-11ea-89c9-a16daa0220c4.png)

**Keycloak returning valid tokens after login success:**

![image](https://user-images.githubusercontent.com/32032862/78433970-60f37880-76d2-11ea-9406-d8bb9d0f87cb.png)

**Login to keycloak success but Viewer not adding Authentication header to the AJAX call to orthanc:**

![image](https://user-images.githubusercontent.com/32032862/78434466-8b453600-76d2-11ea-8d32-dbf1fc612870.png)
